Apps/Sandcastle/gallery/Cesium OSM Buildings Styling.html

```diff
@@ -0,0 +1,260 @@
+        var buildings = Cesium.createOsmBuildings();
+        viewer.scene.primitives.add(buildings);
+
+        // Set the initial camera to look at Hong Kong
+        viewer.scene.camera.flyTo({
+          destination: Cesium.Cartesian3.fromDegrees(-122.3472, 47.598, 370),
+          orientation: {
+            heading: Cesium.Math.toRadians(10),
+            pitch: Cesium.Math.toRadians(-10),
+          },
+        });
+
+        // Styling functions
+        function colorByHeight() {
+          buildings.style = new Cesium.Cesium3DTileStyle({
+            color: {
+              conditions: [
+                [
+                  "${feature['cesium#estimatedHeight']} >= 100",
+                  "color('#581845')",
+                ],
+                [
+                  "${feature['cesium#estimatedHeight']} >= 50",
+                  "color('#900C3F')",
+                ],
+                [
+                  "${feature['cesium#estimatedHeight']} >= 25",
+                  "color('#C70039')",
+                ],
+                [
+                  "${feature['cesium#estimatedHeight']} >= 10",
+                  "color('#FF5733')",
+                ],
+                ["true", "color('#FFC300')"],
+              ],
+            },
+          });
+        }
+
+        function colorByLatitude() {
+          buildings.style = new Cesium.Cesium3DTileStyle({
+            defines: {
+              lagitudeRadians: "radians(${feature['cesium#latitude']})",
+            },
+            color: {
+              conditions: [
+                ["${lagitudeRadians} >= 0.83117", "color('purple')"],
+                ["${lagitudeRadians} >= 0.83112", "color('red')"],
+                ["${lagitudeRadians} >= 0.83107", "color('orange')"],
+                ["${lagitudeRadians} >= 0.83102", "color('yellow')"],
+                ["${lagitudeRadians} >= 0.83097", "color('lime')"],
+                ["${lagitudeRadians} >= 0.83092", "color('cyan')"],
+                ["true", "color('blue')"],
+              ],
+            },
+          });
+        }
+
+        function colorByMaterial() {
+          buildings.style = new Cesium.Cesium3DTileStyle({
+            defines: {
+              material: "${feature['building:material']}",
+            },
+            color: {
+              conditions: [
+                ["${material} === null", "color('white')"],
+                ["${material} === undefined", "color('white')"],
+                ["${material} === 'glass'", "color('skyblue', 0.5)"],
+                ["${material} === 'concrete'", "color('grey')"],
+                ["${material} === 'brick'", "color('indianred')"],
+                ["${material} === 'stone'", "color('lightslategrey')"],
+                ["${material} === 'metal'", "color('lightgrey')"],
+                ["${material} === 'steel'", "color('lightsteelblue')"],
+                ["true", "color('white')"],
+              ],
+            },
+          });
+        }
```

Comment:
This is super cool - I think we should make this the default option, and update the Sandcastle's image to be this one.

Can you also add a comment here explaining that the check for `null` and `undefined` are there because not all buildings have these properties? And that allows you to set the color in those cases.

Reply:
Also, is both `null` and `undefined` necessary? Does it work if you just do one of them?